packages/optimizer/lib/DomTransformer.js

```diff
@@ -49,28 +93,46 @@ class DomTransformer {
    * @param {Object} params - a dictionary containing transformer specific parameters.
    */
   transformTree(tree, params) {
+    params = params || {};
+    log.verbose(params.verbose || false);
     const sequence = (promise, transformer) => {
       return promise.then(() => {
-        const result = Promise.resolve(transformer.transform(tree, params));
-        return result;
+        // not all transformers return a promise
+        return Promise.resolve(transformer.transform(tree, params, this.log_));
       });
     };
-    return this._transformers.reduce(sequence, Promise.resolve());
+    return this.transformers_.reduce(sequence, Promise.resolve());
   }
 
   /**
    * Set the config.
    * @param {Object} config - The config.
+   * @param {boolean} config.verbose - true if verbose mode should be enabled.
```

Comment:
Seems like we are missing the `config.validAmp` parameter here.

Reply:
done